### ISSUE TITLE
Installing gems locally is very slow

### ISSUE THREAD

**Issue by sullerandras** [rubygems/rubygems#1923]
My current problem is that upgrading rubygems 2.6.8 to 2.6.12 made installing gems locally very slow. It was about 20 seconds before, now it is about 8 minutes!

I have a rather big gem which is shared across many applications. During development i often install this gem into the local gemset to try it in one of the applications. I use these commands to locally install the gem:
```
rm *.gem
gem build *.gemspec
gem install *.gem
```
The first two steps are quick, the `gem install` takes ages and it uses 100% cpu during the whole time.

Profiling `gem install` revealed that it is spending lots of time in `Gem::Package#verify_gz` (i used `ruby -rprofile -S gem install -V *.gem` to profile).
After some investigation it turned out that `Gem::Source::Local#load_specs` is called for every dependency of my gem and `load_specs` will load and verify the locally generated gem file every time because of the `Dir["*.gem"]` pattern!

A workaround i found is to move the generated gem file into a different folder before installing it:
```
rm *.gem
rm tmp/*.gem
gem build *.gemspec
mv *.gem tmp/
gem install tmp/*.gem
```

This issue is related to:

  - [ ] Network problems
  - [x] Installing a library
  - [ ] Publishing a library
  - [x] The command line `gem`
  - [ ] Other

Here are my current environment details:

```
$ gem env version
2.6.12
```

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

**Comment by segiddins** [comment]
Can you please share a link to a gem we can use to profile this performance regression? Thanks!

**Comment by sullerandras** [comment]
Created a gem: https://github.com/sullerandras/demonstrate-rubygems-issue-1923
Building that gem takes about 60 seconds. The resulting gem file is about 2.9 MB.
If i remove `lib/random.rb` file it goes down to about 10 seconds and the gem file is about 4 kB.

So the gem file should be significantly big to notice the slowness and there must be dependencies in the gemspec.

Hope this helps!

**Comment by segiddins** [comment]
Thanks, reproduced and I have a few ideas I'll try out

**Comment by segiddins** [comment]
See if https://github.com/rubygems/rubygems/pull/1925 fixes things!

**Comment by sullerandras** [comment]
Thanks for the quick fix, #1925 made it quick for me.
